src/modules/paymentProcessor/PP_Queue_v1.sol

```diff
@@ -0,0 +1,1081 @@
+        // Delete the field.
+        delete _unclaimableAmountsForRecipient[client_][token_][receiver_];
+
+        // Transfer amount to treasury. Call has to succeed otherwise no state
+        // change.
+        IERC20(token_).safeTransferFrom(
+            address(this), _failedOrdersTreasury, amount
+        );
+
+        emit TokensReleased(receiver_, address(token_), amount);
+        emit UnclaimableAmountClaimedToTreasury(
+            receiver_, _failedOrdersTreasury, amount, _msgSender()
+        );
+    }
+
+    /// @inheritdoc IPP_Queue_v1
+    function cancelPaymentOrderThroughQueueId(
+        uint orderId_,
+        IERC20PaymentClientBase_v2 client_
+    )
+        external
+        virtual
+        onlyModuleRole(QUEUE_OPERATOR_ROLE)
+        returns (bool success_)
+    {
+        // Validate that the order exists for the given queue ID and client.
+        if (!_orderExists(orderId_, client_)) {
+            revert Module__PP_Queue_InvalidOrderId(address(client_), orderId_);
+        }
+
+        // Get the order to be cancelled
+        QueuedOrder storage order = _orders[address(client_)][orderId_];
+
+        // Check if the order is in a valid state for cancellation (must be PENDING).
+        if (order.state_ != RedemptionState.PENDING) {
+            revert Module__PP_Queue_InvalidState();
+        }
+
+        // Check if the client has enough balance to cancel the order, otherwise revert.
+        if (
+            IERC20(order.order_.paymentToken).balanceOf(order.client_)
+                < order.order_.amount
+        ) {
+            revert Module__PP_Queue_InvalidAmount(order.order_.amount);
+        }
+
+        // Update the order state to CANCELLED.
+        _updateOrderState(orderId_, address(client_), RedemptionState.CANCELLED);
+
+        // Remove the cancelled order from the queue.
+        _removeFromQueue(orderId_, address(client_));
+
+        // Try to transfer the amount to the treasury.
+        success_ = _tryPaymentTransfer(
+            order.order_.paymentToken,
+            order.client_,
+            _cancelledOrdersTreasury,
+            order.order_.amount,
+            false // don't collect protocol fee when cancelling order
+        );
+    }
+
+    // -------------------------------------------------------------------------
+    // Internal Functions
+
+    ///	@notice	Processes the next payment order in the queue.
+    ///	@return	success_ True if a payment was processed.
+    function _processNextOrder(address client_)
+        internal
+        virtual
+        clientIsValid(client_)
+        returns (bool success_)
+    {
+        uint firstId = _queue[client_].getNextId(LinkedIdList._SENTINEL);
+        if (firstId == LinkedIdList._SENTINEL) {
+            return false;
+        }
+
+        QueuedOrder storage order = _orders[client_][firstId];
+
+        // Skip if order is not in PENDING state.
+        if (order.state_ != RedemptionState.PENDING) {
```

Comment:
1. afaics thsi should never happen, as each time the state is changed from pending, the order is removed from the queue. So perhaps not fail silently here? 
2. If I am wrong and this happens, why return `false`? This will cause the loop in _executePaymentQueue to exit, and so the operator will need to call processPayments a second time to process the rest of the queue.